records/2022-07-20.security.prefix-api-token.md

```diff
@@ -0,0 +1,39 @@
+# TITLE
+
+Date: 2022-07-20
+
+## Status
+
+**DRAFT**
+
+## Context
+
+The mishandling of application secrets has lead to many security incidents in companies big and small. Given the inherent risk in how users store these secrets, it is important that we provide mechanisms to protect our users from doing so improperly.  Taking a cue from [Github](https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/), [Slack](https://api.slack.com/authentication/token-types), and [Stripe](https://stripe.com/docs/api/authentication), we see that token prefixes are an effective technique to help us ensure our tokens don't end up in public repositories.
+
+By combining a known prefix with Github's [secret scanning program](https://docs.github.com/en/developers/overview/secret-scanning-partner-program), we can be alerted if one of our users mistakenly stores their API keys in the open and we can take action.
+
+## Options
+
+- Add a known prefix to our API keys in order to be able to easily identify them
+- Register with Github's secret scanning program and create an endpoint for them to report issues back to us
```

Comment:
That part might be on the SRE team. Let's ask them if rely on them for that. Good to have in the ADR to explain the flow. 👍  CC @CalvinRodo

Reply:
Hey @CalvinRodo, do you think hooking into to Github's secret scanning program is something the SRE team can take on? (https://docs.github.com/en/developers/overview/secret-scanning-partner-program)